<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1366</width>
    <height>768</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Stereo Camera Calibration with QT and openCV - by AbsurdePhoton - www.absurdephoton.fr</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QProgressBar" name="progres">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>265</y>
      <width>871</width>
      <height>23</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>Number of stereo pairs</string>
    </property>
    <property name="styleSheet">
     <string notr="true">QProgressBar {
     border: 2px solid grey;
     border-radius: 5px;
     background-color: #DFDFDF;
	 text-align: center;
 }

 QProgressBar::chunk {
    background-color: qlineargradient(spread:pad, x1:0, y1:0, x2:0, y2:1, stop:0 rgba(0,192, 255, 255), stop:1 rgba(0, 192, 192, 255));
     width: 20px;
 }</string>
    </property>
    <property name="value">
     <number>0</number>
    </property>
   </widget>
   <widget class="QPushButton" name="load">
    <property name="geometry">
     <rect>
      <x>940</x>
      <y>160</y>
      <width>151</width>
      <height>61</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Launch calibration using the parameters defined in the GUI.&lt;/p&gt;&lt;p&gt;Results can be saved in XML files (left, right and stereo)&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
    <property name="text">
     <string>CALIBRATE</string>
    </property>
    <property name="icon">
     <iconset>
      <normalon>:/icons/compute.png</normalon>
     </iconset>
    </property>
    <property name="iconSize">
     <size>
      <width>32</width>
      <height>32</height>
     </size>
    </property>
   </widget>
   <widget class="QLabel" name="droite">
    <property name="geometry">
     <rect>
      <x>350</x>
      <y>0</y>
      <width>331</width>
      <height>251</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>Source RIGHT image</string>
    </property>
    <property name="frameShape">
     <enum>QFrame::Panel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Sunken</enum>
    </property>
    <property name="lineWidth">
     <number>2</number>
    </property>
    <property name="text">
     <string>Source right</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="result_right">
    <property name="geometry">
     <rect>
      <x>490</x>
      <y>340</y>
      <width>471</width>
      <height>361</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>Processed RIGHT image</string>
    </property>
    <property name="frameShape">
     <enum>QFrame::Panel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Sunken</enum>
    </property>
    <property name="lineWidth">
     <number>2</number>
    </property>
    <property name="text">
     <string>Result left</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="result_left">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>340</y>
      <width>471</width>
      <height>361</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>Processed LEFT image</string>
    </property>
    <property name="frameShape">
     <enum>QFrame::Panel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Sunken</enum>
    </property>
    <property name="lineWidth">
     <number>2</number>
    </property>
    <property name="text">
     <string>Result right</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="gauche">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>0</y>
      <width>331</width>
      <height>251</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>Source LEFT image</string>
    </property>
    <property name="frameShape">
     <enum>QFrame::Panel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Sunken</enum>
    </property>
    <property name="lineWidth">
     <number>2</number>
    </property>
    <property name="text">
     <string>Source left</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QProgressBar" name="failed">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>305</y>
      <width>871</width>
      <height>23</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>Number of ERRORS</string>
    </property>
    <property name="styleSheet">
     <string notr="true">QProgressBar {
     border: 2px solid grey;
     border-radius: 5px;
     background-color: #DFDFDF;
	 text-align: center;
 }

 QProgressBar::chunk {
    background-color: qlineargradient(spread:pad, x1:0, y1:0, x2:0, y2:1, stop:0 rgba(255, 0, 0, 255), stop:1 rgba(128, 0, 0, 255));
     width: 20px;
 }</string>
    </property>
    <property name="value">
     <number>0</number>
    </property>
   </widget>
   <widget class="QLCDNumber" name="lcd_good">
    <property name="geometry">
     <rect>
      <x>897</x>
      <y>260</y>
      <width>64</width>
      <height>31</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>Number of GOOD stereo pairs</string>
    </property>
    <property name="styleSheet">
     <string notr="true">QLCDNumber{ 
    background-color: rgb(64, 64,64);
}</string>
    </property>
    <property name="digitCount">
     <number>3</number>
    </property>
   </widget>
   <widget class="QLCDNumber" name="lcd_bad">
    <property name="geometry">
     <rect>
      <x>897</x>
      <y>300</y>
      <width>64</width>
      <height>31</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>Number of BAD stereo pairs</string>
    </property>
    <property name="styleSheet">
     <string notr="true">QLCDNumber{ 
    background-color: rgb(64, 64,64);
}</string>
    </property>
    <property name="digitCount">
     <number>3</number>
    </property>
   </widget>
   <widget class="QRadioButton" name="radioButton_checkerboards">
    <property name="geometry">
     <rect>
      <x>710</x>
      <y>20</y>
      <width>131</width>
      <height>22</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>Process CHECKERBOARDS</string>
    </property>
    <property name="text">
     <string>Chec&amp;kerboards</string>
    </property>
    <property name="checked">
     <bool>true</bool>
    </property>
    <attribute name="buttonGroup">
     <string notr="true">type</string>
    </attribute>
   </widget>
   <widget class="QRadioButton" name="radioButton_circles_symmetric">
    <property name="geometry">
     <rect>
      <x>710</x>
      <y>50</y>
      <width>171</width>
      <height>22</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>Process SYMMETRIC CIRCLES boards</string>
    </property>
    <property name="text">
     <string>Circ&amp;les (symmetric)</string>
    </property>
    <attribute name="buttonGroup">
     <string notr="true">type</string>
    </attribute>
   </widget>
   <widget class="QSpinBox" name="spinBox_width">
    <property name="geometry">
     <rect>
      <x>710</x>
      <y>160</y>
      <width>71</width>
      <height>27</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>Checkerboard : number of inner corners (horizontal)
Symmetric circles : number of columns
Asymmetric circles : number of columns ('n' circles per shifted row)</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
    <property name="value">
     <number>6</number>
    </property>
   </widget>
   <widget class="QSpinBox" name="spinBox_height">
    <property name="geometry">
     <rect>
      <x>810</x>
      <y>160</y>
      <width>71</width>
      <height>27</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>Checkerboard : number of inner corners (vertical)
Circles : number of rows</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
    <property name="value">
     <number>9</number>
    </property>
   </widget>
   <widget class="QSpinBox" name="spinBox_size">
    <property name="geometry">
     <rect>
      <x>740</x>
      <y>196</y>
      <width>71</width>
      <height>27</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>Size in mm of each element :
* square size (checkerboard)
* circle diameter (circles)</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
    <property name="value">
     <number>20</number>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>760</x>
      <y>130</y>
      <width>67</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>Board :</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>790</x>
      <y>165</y>
      <width>16</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>X</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>820</x>
      <y>200</y>
      <width>31</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>mm</string>
    </property>
   </widget>
   <widget class="QFrame" name="frame">
    <property name="geometry">
     <rect>
      <x>700</x>
      <y>10</y>
      <width>191</width>
      <height>101</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::Panel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Sunken</enum>
    </property>
    <property name="lineWidth">
     <number>2</number>
    </property>
    <widget class="QRadioButton" name="radioButton_circles_asymmetric">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>70</y>
       <width>171</width>
       <height>22</height>
      </rect>
     </property>
     <property name="toolTip">
      <string>Process ASYMMETRIC CIRCLES boards</string>
     </property>
     <property name="text">
      <string>Circles (asy&amp;mmetric)</string>
     </property>
     <attribute name="buttonGroup">
      <string notr="true">type</string>
     </attribute>
    </widget>
   </widget>
   <widget class="QFrame" name="frame_2">
    <property name="geometry">
     <rect>
      <x>700</x>
      <y>120</y>
      <width>191</width>
      <height>111</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::Panel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Sunken</enum>
    </property>
    <property name="lineWidth">
     <number>2</number>
    </property>
   </widget>
   <widget class="QFrame" name="frame_3">
    <property name="geometry">
     <rect>
      <x>900</x>
      <y>10</y>
      <width>231</width>
      <height>101</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::Panel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Sunken</enum>
    </property>
    <property name="lineWidth">
     <number>2</number>
    </property>
    <widget class="QDoubleSpinBox" name="sensor_width">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>50</y>
       <width>69</width>
       <height>27</height>
      </rect>
     </property>
     <property name="toolTip">
      <string>Camera sensor horizontal size (in mm)</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
     <property name="value">
      <double>17.300000000000001</double>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="sensor_height">
     <property name="geometry">
      <rect>
       <x>110</x>
       <y>50</y>
       <width>69</width>
       <height>27</height>
      </rect>
     </property>
     <property name="toolTip">
      <string>Camera sensor vertical size (in mm)</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
     <property name="value">
      <double>13.000000000000000</double>
     </property>
    </widget>
    <widget class="QLabel" name="label_5">
     <property name="geometry">
      <rect>
       <x>90</x>
       <y>56</y>
       <width>16</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>X</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_6">
     <property name="geometry">
      <rect>
       <x>190</x>
       <y>55</y>
       <width>31</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>mm</string>
     </property>
    </widget>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>970</x>
      <y>20</y>
      <width>67</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>Sensor :</string>
    </property>
   </widget>
   <widget class="QFrame" name="frame_4">
    <property name="geometry">
     <rect>
      <x>900</x>
      <y>120</y>
      <width>231</width>
      <height>111</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::Panel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Sunken</enum>
    </property>
    <property name="lineWidth">
     <number>2</number>
    </property>
    <widget class="QRadioButton" name="radioButton_alternate">
     <property name="geometry">
      <rect>
       <x>110</x>
       <y>13</y>
       <width>110</width>
       <height>21</height>
      </rect>
     </property>
     <property name="toolTip">
      <string>Select several image files (with  &lt;SHIFT&gt; and &lt;CTRL&gt;) : 
* 2 files for one stereo pair
* the pair is composed of 2 separate images : first file = LEFT, second file = RIGHT
* the total number of files must be EVEN</string>
     </property>
     <property name="text">
      <string>Alter&amp;nate</string>
     </property>
     <attribute name="buttonGroup">
      <string notr="true">buttonGroup</string>
     </attribute>
    </widget>
    <widget class="QRadioButton" name="radioButton_MPO">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>13</y>
       <width>71</width>
       <height>21</height>
      </rect>
     </property>
     <property name="toolTip">
      <string>Select several MPO stereo pair files (select multiple files with &lt;SHIFT&gt; and &lt;CTRL&gt;)</string>
     </property>
     <property name="text">
      <string>MPO</string>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
     <attribute name="buttonGroup">
      <string notr="true">buttonGroup</string>
     </attribute>
    </widget>
   </widget>
   <widget class="QLabel" name="label_7">
    <property name="geometry">
     <rect>
      <x>970</x>
      <y>580</y>
      <width>221</width>
      <height>121</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::Box</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Sunken</enum>
    </property>
    <property name="lineWidth">
     <number>3</number>
    </property>
    <property name="text">
     <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p align=&quot;center&quot;&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;Stereo Camera Calibration&lt;/span&gt;&lt;/p&gt;&lt;p align=&quot;center&quot;&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;with OpenCV&lt;/span&gt;&lt;/p&gt;&lt;p align=&quot;center&quot;&gt;by &lt;span style=&quot; color:#d89d1a;&quot;&gt;AbsurdePhoton&lt;/span&gt;&lt;/p&gt;&lt;p align=&quot;center&quot;&gt;&lt;a href=&quot;https::/www.absurdephoton.fr&quot;&gt;&lt;span style=&quot; text-decoration: underline; color:#0000ff;&quot;&gt;www.absurdephoton.fr&lt;/span&gt;&lt;/a&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
    <property name="openExternalLinks">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QLabel" name="label_8">
    <property name="geometry">
     <rect>
      <x>1160</x>
      <y>610</y>
      <width>71</width>
      <height>61</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>AbsurdePhoton - www.absurdephoton.fr</string>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="pixmap">
     <pixmap resource="resources.qrc">:/icons/absurdephoton.png</pixmap>
    </property>
   </widget>
   <widget class="QPushButton" name="Abort">
    <property name="geometry">
     <rect>
      <x>990</x>
      <y>263</y>
      <width>121</width>
      <height>61</height>
     </rect>
    </property>
    <property name="text">
     <string> Abort</string>
    </property>
    <property name="icon">
     <iconset>
      <normalon>:/icons/stop.png</normalon>
     </iconset>
    </property>
    <property name="iconSize">
     <size>
      <width>32</width>
      <height>32</height>
     </size>
    </property>
   </widget>
   <zorder>frame_4</zorder>
   <zorder>frame</zorder>
   <zorder>frame_2</zorder>
   <zorder>progres</zorder>
   <zorder>load</zorder>
   <zorder>droite</zorder>
   <zorder>result_right</zorder>
   <zorder>result_left</zorder>
   <zorder>gauche</zorder>
   <zorder>failed</zorder>
   <zorder>lcd_good</zorder>
   <zorder>lcd_bad</zorder>
   <zorder>radioButton_checkerboards</zorder>
   <zorder>radioButton_circles_symmetric</zorder>
   <zorder>spinBox_width</zorder>
   <zorder>spinBox_height</zorder>
   <zorder>spinBox_size</zorder>
   <zorder>label</zorder>
   <zorder>label_2</zorder>
   <zorder>label_3</zorder>
   <zorder>frame_3</zorder>
   <zorder>label_4</zorder>
   <zorder>label_7</zorder>
   <zorder>label_8</zorder>
   <zorder>Abort</zorder>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources>
  <include location="resources.qrc"/>
 </resources>
 <connections/>
 <buttongroups>
  <buttongroup name="type"/>
  <buttongroup name="buttonGroup"/>
 </buttongroups>
</ui>
